src/autoupdate.c

```diff
@@ -29,56 +29,76 @@
 
 #include "swupd.h"
 
-static void print_help(const char *name)
+static void print_help(void)
 {
 	print("Usage:\n");
-	print("   swupd %s [options]\n\n", basename((char *)name));
-	print("Help Options:\n");
-	print("   -h, --help              Show help options\n");
+	print("   swupd autoupdate [OPTION...]\n\n");
+
+	global_print_help();
+
+	print("Options:\n");
 	print("       --enable            enable autoupdates\n");
 	print("       --disable           disable autoupdates\n");
 	print("\n");
 }
 
-static int enable;
-static int disable;
+#define FLAG_ENABLE 2000
+#define FLAG_DISABLE 2001
+static bool cmdline_option_enable;
+static bool cmdline_option_disable;
 
 static const struct option prog_opts[] = {
 	{ "help", no_argument, 0, 'h' },
-	{ "enable", no_argument, &enable, 1 },
-	{ "disable", no_argument, &disable, 1 },
-	{ 0, 0, 0, 0 }
+	{ "enable", no_argument, 0, FLAG_ENABLE },
+	{ "disable", no_argument, 0, FLAG_DISABLE },
+};
+
+static bool parse_opt(int opt, UNUSED_PARAM char *optarg)
+{
+	switch (opt) {
+	case 'h':
+		print_help();
+		exit(EXIT_SUCCESS);
+	case FLAG_ENABLE:
+		cmdline_option_enable = true;
```

Comment:
ups, one more thing, instead of directly assigning `true` you need to use the `optarg_to_bool` macro like this `cmdline_option_enable = optarg_to_bool(optarg);` otherwise if those values are added to the config file they won't be parsed correctly.

Reply:
Good catch! Thanks Done..